lib/libzfs/libzfs_sendrecv.c

```diff
@@ -2725,10 +2725,137 @@ recv_incremental_replication(libzfs_handle_t *hdl, const char *tofs,
 	return (needagain);
 }
 
+/*
+ * Update stream_avl based on override properties:
+ * 1. remove properties from "props" e "snapprops" nvlists if they are present
+ * as boolean values in override_props
+ * 2. add or overwrite string properties from override_props only on the
+ * ancestor "props" and remove them from child "props" and "snapprops" so they
+ * are (possibly) inherited.
+ */
+static int
+props_override(libzfs_handle_t *hdl, avl_tree_t *stream_avl, nvlist_t *props,
+    recvflags_t *flags, const char *errbuf)
+{
+	nvlist_t *noprops;
```

Comment:
The nesting can be reduced by doing `if (nvlist not present) continue; <handle nvlist>` rather than `if (nvlist present) {<handle nvlist>}`.  In any case, this particular variable doesn't appear deep in the nesting, so it can afford to have a longer name.  I understand the desire to avoid overly-verbose names, though; this is the only one whose name I really have issue with.